### DISCUSSION TITLE
Specify matrix block type

### DISCUSSION THREAD

**Discussion by Tam** [engram-design/FeedMe#891] [Ideas]
### Description
We're rebuilding a craft site and need to import matrix blocks from the old site to the new, while maintaining their order. As far as I can see there's no way of specifying the block type to use when importing. Our data is structured like this (via element api):

```json
{
  "data": [
    {
      "layers": [
        {
          "type": "blockType",
          "all": "fields",
          "for": "all",
          "block": "types",
        }
      ]
    }
  ]
}
```

Each `layer` item has all the possible fields for each block type. Fields that don't appear on that block type are `null`.

### Steps to reproduce

1. Create a Craft 2 site that has some matrix block in it.
2. Re-build the entire site from scratch in Craft 3.
3. Try to import the matrix blocks, while maintaining their order.
5. Fail.
8. Try not to cry.
13. Cry a lot.

### Additional info

- Plugin version: 3.0.4
- Craft version: 3.1.16

**Comment by engram-design** [comment]
Its correct that you can't select the block type when mapping, because the fields change depend on what block type you want - therefore, a potentially different set of mapping. It'd be like trying to set the entry type or section of an entry when trying to import it - Feed Me wouldn't know what to do!

You should be able to import your content in the correct order via:
```
{
  "entries": [
  {
    "title": "Title 1-1",
    "MatrixField": {
      "MatrixBlock": [
        {
          "HeadingSize": "h1",
          "HeadingText": "This is an H1 tag"
        },
        { 
          "Copy": "<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p>"
        },
        {
          "HeadingSize": "h2",
          "HeadingText": "This is an H2 tag"
        },
        {
          "Images": [
            "img_fjords.jpg"
          ]
        }
      ]
    }
  }
}
```

Where the above shows importing an entry, with a Matrix field, with 3 block types, Heading, Image and Copy, and these should import in the order specified, and for the type setup per mapping. Note the nodes names are just for illustrative purposes, but if you're going from C2 > C3, it might be worth keeping them the same as their counterparts.

**Comment by Tam** [comment]
That would work, except two of the blocks we've got share the same fields (same field names) but are displayed differently. I guess the only workaround is to change the name of at least one of the fields.

**Comment by engram-design** [comment]
Yep, good point. Marking as a feature request for now.
